packages/dotnet/src/lib/core/dotnet.client.ts

```diff
@@ -24,160 +23,161 @@ import { LoadedCLI } from './dotnet.factory';
-      const paramString = parameters ? getParameterString(parameters) : '';
-      cmd = `${cmd} ${paramString}`;
+      params.push(...getParameterArrayStrings(parameters));
     }
     if (publishProfile) {
-      cmd = `${cmd} -p:PublishProfile=${publishProfile}`;
+      params.push(`-p:PublishProfile=${publishProfile}`);
     }
     if (extraParameters) {
-      cmd = `${cmd} ${extraParameters}`;
+      params.push(`${extraParameters}`);
     }
-    return this.logAndExecute(cmd);
+    return this.logAndExecute(params);
   }
 
-  installTool(tool: string): void {
-    const cmd = `${this.cliCommand.command} tool install ${tool}`;
+  installTool(tool: string): Buffer {
+    const cmd = [`tool`, `install`, tool];
     return this.logAndExecute(cmd);
   }
 
-  restorePackages(project: string): void {
-    const cmd = `${this.cliCommand.command} restore "${project}"`;
+  restorePackages(project: string): Buffer {
+    const cmd = [`restore`, project];
     return this.logAndExecute(cmd);
   }
 
-  restoreTools(): void {
-    const cmd = `${this.cliCommand.command} tool restore`;
+  restoreTools(): Buffer {
+    const cmd = [`tool`, `restore`];
     return this.logAndExecute(cmd);
   }
 
-  format(project: string, parameters?: dotnetFormatOptions): void {
-    let cmd = `${this.cliCommand.command} format "${project}"`;
+  format(project: string, parameters?: dotnetFormatOptions): Buffer {
+    const params = [`format`, project];
     if (parameters) {
       parameters = swapKeysUsingMap(parameters, formatKeyMap);
-      const paramString = parameters ? getParameterString(parameters) : '';
-      cmd = `${cmd} ${paramString}`;
+      params.push(...getParameterArrayStrings(parameters));
     }
-    return this.logAndExecute(cmd);
+    return this.logAndExecute(params);
   }
 
   addProjectToSolution(solutionFile: string, project: string) {
-    const cmd = `${this.cliCommand.command} sln "${solutionFile}" add "${project}"`;
-    this.logAndExecute(cmd);
+    const params = [`sln`, solutionFile, `add`, project];
+    this.logAndExecute(params);
   }
 
   getSdkVersion(): Buffer {
-    const cmd = 'dotnet --version';
-    return this.execute(cmd);
+    return this.execute(['--version']);
   }
 
   printSdkVersion(): void {
-    this.logAndExecute('dotnet --version');
+    this.logAndExecute(['--version']);
   }
 
-  private logAndExecute(cmd: string): void {
-    console.log(`Executing Command: ${cmd}`);
-    execSync(cmd, { stdio: 'inherit', cwd: this.cwd || process.cwd() });
+  private logAndExecute(params: string[]): Buffer {
+    console.log(
+      `Executing Command: ${this.cliCommand.command} ${params.join(', ')}`,
+    );
+    return this.execute(params);
   }
 
-  private execute(cmd: string): Buffer {
-    return execSync(cmd, { cwd: this.cwd || process.cwd() });
+  private execute(params: string[]): Buffer {
+    return spawnSync(this.cliCommand.command, params, {
+      stdio: 'inherit',
```

Comment:
Adding 'inherit' here causes the returned buffers to be empty

Reply:
I have a new solution for this have to discuss with you, I didn't merge it yet. This should resolve the problem:

```ts
  private execute(params: string[]): Buffer {
    const proc = spawnSync(this.cliCommand.command, params, {
      stdio: 'inherit',
      cwd: this.cwd || process.cwd(),
    });

    let result = Buffer.alloc(0);
    if (proc?.stdout) {
      result = Buffer.concat([result, proc.stdout]);
    }
    if (proc?.stderr) {
      result = Buffer.concat([result, proc.stderr]);
    }
    return result;
  }
```

Reply:
This would work, but its fine to duplicate the spawnSync call for this. The idea behind `execute` is that we are doing some kind of processing to the Buffer on our side, and we need it. If we are displaying the outputs of the command to the user, its the wrong method to use.

Reply:
I agree with different calls for spawnSync with different behavior.
Still, we should return the buffer from `stderr` and `stdout` and not the result of spawnSync that has the type `SpawnSyncReturns<Buffer>`, I'll open a new issue for this.